### ISSUE TITLE
Possible source for decompiled sub_110830

### ISSUE THREAD

**Issue by shark0der** [pkmngodev/Unknown6#10]
I saw some strings in `sub_110830` and googled them, leading me to this findings:

[sub_110830](https://github.com/pkmngodev/Unknown6/blob/master/subs/sub_110830.c#L230) looks similar to this:
https://github.com/google/protobuf/blob/v2.6.1/src/google/protobuf/wire_format.cc#L1093

**Comment by gnought** [comment]
It seems it is same as

```
void WireFormat::VerifyUTF8StringFallback(const char* data,
                                          int size,
                                          Operation op,
                                          const char* field_name) {
  if (!IsStructurallyValidUTF8(data, size)) {
    const char* operation_str = NULL;
    switch (op) {
      case PARSE:
        operation_str = "parsing";
        break;
      case SERIALIZE:
        operation_str = "serializing";
        break;
      // no default case: have the compiler warn if a case is not covered.
    }
    string quoted_field_name = "";
    if (field_name != NULL) {
      quoted_field_name = StringPrintf(" '%s'", field_name);
    }
    // no space below to avoid double space when the field name is missing.
    GOOGLE_LOG(ERROR) << "String field" << quoted_field_name << " contains invalid "
               << "UTF-8 data when " << operation_str << " a protocol "
               << "buffer. Use the 'bytes' type if you intend to send raw "
               << "bytes. ";
  }
}
```

**Comment by myrdyr** [comment]
Similarily, sub_1B6FC0 is std::logic_error, sub_19B6C4 std::basic_string etc.
It's possible to remove a lot of the subs and replace their names with the proper standard library call.

**Comment by gurupras** [comment]
Noob question :(
Doesn't objdump tell you function names and addresses?

**Comment by shark0der** [comment]
Binary is stripped. No function names, no debug symbols

**Comment by ChronoXNL** [comment]
@gurupras : no, as the function names pretty much get lost once compiled (they do not serve a purpose in compiled code, they just get 'tagged' as a symbol

**Comment by gurupras** [comment]
I'm confused then. When I run objdump on the .so, I get functions like:  

    00049610 <UnityPluginLoad>:
    00049614 <UnityPluginUnload>:

I'll take your word for it that this isn't very useful

**Comment by shark0der** [comment]
Some unity generic functions may have names. The important code is not. @repomods please delete irrelevant comments

**Comment by rickytesting2016** [comment]
One more stupid question (in case if we didn't think/try it before...)
Did we try to decompile previous version of PokemonGo, even like maybe alpha/beta/0.01 version on Android/iOS.
Assuming unknown6 exists long time ago, during init stage of development / testing / beta...they *may* not strip out so many things for debugging....or at least it may remain more strings / information?

**Comment by shark0der** [comment]
It's up to you to find out. If you do find something - open a new issue.

**Comment by rickytesting2016** [comment]
Just an idea as i haven't see anyone try it in reddit / github, but I'm not going to work on it LOL

**Comment by Jean-Daniel** [comment]
@gurupras objdump shows you exported symbols which must not be stripped as they are used by the dynamic linker.
Other symbols are stripped. Moreover, the compiler can also inline some code, and so completely remove the function from the output library (the inlined function appears as being part of the calling function, and you don't even know it was a independent function in the first place, so you can't got the function name)

**Comment by Dingdanipp** [comment]
i can work on it. There is a symbol recovered binary https://dl.dropboxusercontent.com/u/83439509/armeabi-v7a.zip on wiki. The link is broken. What is that? Who still has that file?

**Comment by shark0der** [comment]
http://www.apkmirror.com/apk/niantic-inc/pokemon-go/#variants

**Comment by Phantisy** [comment]
i had the same idea about comparing them to the older nets versions and was trying to chat with users in the discord about that. I already found the original Japan beta and the bets version from back in may. I shared them with in the discord chat user To (I this was his name). If anyone else wants them I'll gladly share the links.

**Comment by Dingdanipp** [comment]
yes i need that @Phantisy I am search all subs here against anything possible

**Comment by Phantisy** [comment]
PokemonGo Field Test version 0.23.2:
https://mega.nz/#!E9pRERzK!Oa90YB5cBcXachZoMFJBXVC9Wx8KQpFcyQ6pIvzoUIQ
PokemonGo Field Test Version (Japan) 0.13.5:
https://mega.nz/#!8o4wAaoD!OiNmEpQj_5f3VE26XDq38DI8byDG8C_0YyydOdSMw68

**Comment by Dingdanipp** [comment]
@Phantisy  no... i miss the second one (0.13.5). link is broken

**Comment by Phantisy** [comment]
0.13.5 https://mega.nz/#!h44gGT6R!OiNmEpQj_5f3VE26XDq38DI8byDG8C_0YyydOdSMw68
0.32.2 https://mega.nz/#!E9pRERzK!Oa90YB5cBcXachZoMFJBXVC9Wx8KQpFcyQ6pIvzoUIQ

**Comment by 4rthurRousseau** [comment]
@rickytesting2016 "Diff-ing different client versions does NOT produce meaningful results. Nothing to compare"

**Comment by Dingdanipp** [comment]
no luck. not much symbols in all previous version. I use IDA Pro with Kam1n0. not sure if there some tricks to load more symbols.
Here is the search result for subs in this repo:

https://www.docdroid.net/kxjw6ai/pkmg.pdf.html

The size of the pie indicates similarity on assembly code

**Comment by shark0der** [comment]
@4rthurRousseau He was not trying to diff. He was attempting to get debug symbols.

**Comment by Dingdanipp** [comment]
@4rthurRousseau That's what i did and reported. There does exist diffs; but not meaningful. See https://www.docdroid.net/kxjw6ai/pkmg.pdf.html here

**Comment by 4rthurRousseau** [comment]
I misunderstood what he was trying to do.
By the way, thanks for the previous versions and the subs report.

**Comment by Dingdanipp** [comment]
anyone knows if there exists other armv7 binaries related to proto buff? Probably we can find a match of some generated functions.

**Comment by ChronoXNL** [comment]
I would say Ingress is using similar / the same perhaps? It's the other app from Niantic that has possibly the same type of encryption.
